modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/WikiListPagesDisplayContext.java

```diff
@@ -590,6 +633,40 @@ else if (navigation.equals("recent-changes")) {
 		}
 	}
 
+	private boolean _canViewPendingStatus(
+		ThemeDisplay themeDisplay, WikiPage page) {
+
+		PermissionChecker permissionChecker =
+			themeDisplay.getPermissionChecker();
+
+		if (permissionChecker.isContentReviewer(
+				themeDisplay.getCompanyId(), themeDisplay.getScopeGroupId())) {
+
+			return true;
+		}
+
+		WikiPage lastPage = null;
+
+		try {
+			lastPage = WikiPageLocalServiceUtil.getPage(
+				page.getResourcePrimKey(), false);
+		}
+		catch (PortalException portalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(portalException);
+			}
+		}
+
+		if ((lastPage == null) ||
+			(page.getVersion() >= lastPage.getVersion()) ||
+			(themeDisplay.getUserId() == lastPage.getStatusByUserId())) {
+
+			return true;
+		}
+
+		return false;
+	}
+
```

Comment:
Doesn't this logic belong to WikiPageModelResourcePermissionWrapper? This logic should be invoked any time a user is trying to access a wiki page (e.g. through the REST API), not only via the portlet.